sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -47,15 +48,20 @@
  */
 public final class  BinaryData {
     private static final ClientLogger LOGGER = new ClientLogger(BinaryData.class);
+
+    private static JsonSerializer defaultJsonSerializer;
+    private static byte[] EMPTY_DATA = new byte[0];
     private final byte[] data;
 
     /**
-     * Create instance of {@link BinaryData} given the data.
+     * Create instance of {@link BinaryData} given the data. If {@code null} value is provided , it will be converted
+     * into empty byte array.
      * @param data to represent as bytes.
-     * @throws NullPointerException If {@code data} is null.
      */
     BinaryData(byte[] data) {
-        Objects.requireNonNull(data, "'data' cannot be null.");
+        if (Objects.isNull(data)) {
```

Comment:
May want to check `CoreUtils.isNullOrEmpty`, this may end up copying a 0 length array.

Reply:
we can not use CoreUtils for byte array. But I have added length check.